source/gui/blockAction.py

```diff
@@ -0,0 +1,57 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import config
+from dataclasses import dataclass
+from enum import Enum
+from functools import wraps
+import globalVars
+from typing import Callable
+import ui
+from gui.message import isModalMessageBoxActive
+
+
+@dataclass
+class _Context:
+	blockActionIf: Callable[[], bool]
+	messageString: str
+
+
+class Context(_Context, Enum):
+	SECURE_MODE = (
+		lambda: globalVars.appArgs.secure,
+		# Translators: Reported when an action cannot be performed because NVDA is in a secure screen
+		_("Action unavailable in secure context"),

```

Comment:
Actually there are two use cases:
* NVDA running on secure screens
* NVDA running in normal context but started with the `--secure` command line parameter.
Ideally, it would be more explicit to have two separate messages for these two distinct contexts. But I think it should be cleaner to separate this work in another PR separating the notions of "secure screen context" and "run with `--secure` parameter".

So for the context of the current PR, I am OK with this message.